index.js

```diff
@@ -75,4 +75,28 @@ exports.hash = function(pwd, salt, fn){
   return function(done){
     fn = done;
   }
-};
\ No newline at end of file
+  
+/**
+ * Hashes asynchronously a password with optional `salt`, otherwise
+ * generate a salt for `pass` and invoke `fn(err, salt, hash)`.
+ *
+ * @param {String} password to hash
+ * @param {String} optional salt
+ * @param {Function} callback
+ * @api public
+ */
+exports.hashAsync = function(pwd, salt, fn){
+    var Promise = require('bluebird');
```

Comment:
global Promise is fine IMO, people can polyfill if they really want bluebird. Also the formatting below should match the rest of the project (2 spaces, brace after `hash) {` etc)